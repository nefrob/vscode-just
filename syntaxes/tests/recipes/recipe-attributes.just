# Recipe attributes

[doc]
[confirm]
[linux]
[macos]
[no-cd]
[no-exit-message]
[private]
[no-quiet]
[unix]
[windows]
foo:
    echo "foo"

# Attribute chaining

[no-cd, private]
foo:
    echo "foo"

# Attribute values

[confirm("Are you sure you want to delete everything?")]
delete-everything:
    rm -rf *

[confirm: "Are you sure you want to delete everything?"]
delete-everything:
    rm -rf *


# Recipe groups

[group('lint')]
js-lint:
    echo 'Running JS linter...'

[group('rust recipes')]
[group('lint')]
rust-lint:
    echo 'Running Rust linter...'

[group('lint')]
cpp-lint:
    echo 'Running C++ linter...'