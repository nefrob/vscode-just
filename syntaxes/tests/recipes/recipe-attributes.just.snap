># Recipe attributes
#^^^^^^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>
>[doc]
#^ source.just
# ^^^ source.just support.function.system.just
#    ^^ source.just
>[extension]
#^ source.just
# ^^^^^^^^^ source.just support.function.system.just
#          ^^ source.just
>[confirm]
#^ source.just
# ^^^^^^^ source.just support.function.system.just
#        ^^ source.just
>[linux]
#^ source.just
# ^^^^^ source.just support.function.system.just
#      ^^ source.just
>[macos]
#^ source.just
# ^^^^^ source.just support.function.system.just
#      ^^ source.just
>[no-cd]
#^ source.just
# ^^^^^ source.just support.function.system.just
#      ^^ source.just
>[no-exit-message]
#^ source.just
# ^^^^^^^^^^^^^^^ source.just support.function.system.just
#                ^^ source.just
>[private]
#^ source.just
# ^^^^^^^ source.just support.function.system.just
#        ^^ source.just
>[no-quiet]
#^ source.just
# ^^^^^^^^ source.just support.function.system.just
#         ^^ source.just
>[script]
#^ source.just
# ^^^^^^ source.just support.function.system.just
#       ^^ source.just
>[unix]
#^ source.just
# ^^^^ source.just support.function.system.just
#     ^^ source.just
>[windows]
#^ source.just
# ^^^^^^^ source.just support.function.system.just
#        ^^ source.just
>foo:
#^^^ source.just entity.name.function.just
#   ^ source.just keyword.operator.recipe.end.just
>    echo "foo"
#^^^^^^^^^^^^^^^ source.just meta.recipe.body.just
>
># Attribute chaining
#^^^^^^^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>
>[no-cd, private]
#^ source.just
# ^^^^^ source.just support.function.system.just
#      ^ source.just
#       ^^^^^^^^ source.just support.function.system.just
#               ^^ source.just
>foo:
#^^^ source.just entity.name.function.just
#   ^ source.just keyword.operator.recipe.end.just
>    echo "foo"
#^^^^^^^^^^^^^^^ source.just meta.recipe.body.just
>
># Attribute values
#^^^^^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>
>[confirm("Are you sure you want to delete everything?")]
#^ source.just
# ^^^^^^^ source.just support.function.system.just
#        ^ source.just
#         ^ source.just string.quoted.double.just string.quoted.double.just
#          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.just string.quoted.double.just
#                                                     ^ source.just string.quoted.double.just
#                                                      ^ source.just
#                                                       ^^ source.just
>delete-everything:
#^^^^^^^^^^^^^^^^^ source.just entity.name.function.just
#                 ^ source.just keyword.operator.recipe.end.just
>    rm -rf *
#^^^^^^^^^^^^^ source.just meta.recipe.body.just
>
>[confirm: "Are you sure you want to delete everything?"]
#^ source.just
# ^^^^^^^ source.just support.function.system.just
#        ^ source.just keyword.operator.attribute.end.just
#         ^ source.just
#          ^ source.just string.quoted.double.just string.quoted.double.just
#           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.just string.quoted.double.just
#                                                      ^ source.just string.quoted.double.just
#                                                       ^^ source.just
>delete-everything:
#^^^^^^^^^^^^^^^^^ source.just entity.name.function.just
#                 ^ source.just keyword.operator.recipe.end.just
>    rm -rf *
#^^^^^^^^^^^^^ source.just meta.recipe.body.just
>
>
># Recipe groups
#^^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>
>[group('lint')]
#^ source.just
# ^^^^^ source.just support.function.system.just
#      ^ source.just
#       ^ source.just string.quoted.single.just string.quoted.single.just
#        ^^^^ source.just string.quoted.single.just
#            ^ source.just string.quoted.single.just
#             ^ source.just
#              ^^ source.just
>js-lint:
#^^^^^^^ source.just entity.name.function.just
#       ^ source.just keyword.operator.recipe.end.just
>    echo 'Running JS linter...'
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.just meta.recipe.body.just
>
>[group('rust recipes')]
#^ source.just
# ^^^^^ source.just support.function.system.just
#      ^ source.just
#       ^ source.just string.quoted.single.just string.quoted.single.just
#        ^^^^^^^^^^^^ source.just string.quoted.single.just
#                    ^ source.just string.quoted.single.just
#                     ^ source.just
#                      ^^ source.just
>[group('lint')]
#^ source.just
# ^^^^^ source.just support.function.system.just
#      ^ source.just
#       ^ source.just string.quoted.single.just string.quoted.single.just
#        ^^^^ source.just string.quoted.single.just
#            ^ source.just string.quoted.single.just
#             ^ source.just
#              ^^ source.just
>rust-lint:
#^^^^^^^^^ source.just entity.name.function.just
#         ^ source.just keyword.operator.recipe.end.just
>    echo 'Running Rust linter...'
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.just meta.recipe.body.just
>
>[group('lint')]
#^ source.just
# ^^^^^ source.just support.function.system.just
#      ^ source.just
#       ^ source.just string.quoted.single.just string.quoted.single.just
#        ^^^^ source.just string.quoted.single.just
#            ^ source.just string.quoted.single.just
#             ^ source.just
#              ^^ source.just
>cpp-lint:
#^^^^^^^^ source.just entity.name.function.just
#        ^ source.just keyword.operator.recipe.end.just
>    echo 'Running C++ linter...'
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.just meta.recipe.body.just
>
># Private aliases and variables
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>
>[private]
#^ source.just
# ^^^^^^^ source.just support.function.system.just
#        ^^ source.just
>alias foo := bar
#^^^^^ source.just keyword.other.reserved.just
#     ^ source.just
#      ^^^ source.just variable.name.alias.just
#         ^ source.just
#          ^^ source.just keyword.operator.assignment.just
#            ^ source.just
#             ^^^ source.just variable.other.just
>
>[private]
#^ source.just
# ^^^^^^^ source.just support.function.system.just
#        ^^ source.just
>baz := 'qux'
#^^^ source.just variable.other.just
#   ^ source.just
#    ^^ source.just keyword.operator.assignment.just
#      ^ source.just
#       ^ source.just string.quoted.single.just string.quoted.single.just
#        ^^^ source.just string.quoted.single.just
#           ^ source.just string.quoted.single.just
>